### ISSUE TITLE
Filter list of connections in companion3 by legacy/not legacy

### ISSUE THREAD

**Issue by josephdadams** [bitfocus/companion#2248]
### Is this a feature relevant to companion itself, and not a module?

- [X] I believe this to be a feature for companion, not a module

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the feature

Filter the list of connections to only show updated modules, or include the legacy ones also. Maybe also list the module version number and/or last date/time its code was updated.

**Comment by Julusian** [comment]
I considered an option to hide legacy modules, but didnt find somewhere good to put it. Anyone fancy making a crude ui mockup in photoshop of where it should go?

Version number would be easy to include. last updated is more challenging, and value could be questionable. I've found a bunch of modules which were last updated by me in february when I was bulk adding some missing properties to package.json files. but perhaps even that date is useful? of course updating for 3.0 will count as an update, even if there are no functional changes

**Comment by josephdadams** [comment]
This is true, sometimes we even do little bumps for dependencies. And an old date last updated doesn’t mean a module is stale, maybe it’s very feature complete and nothing else is needed

**Comment by Julusian** [comment]
With the module plugins system, companion is no longer showing what is a legacy module (it can't really tell the difference)

But maybe the api should report these modules as legacy so that companion can keep doing this? 
Or these modules should simply be deprecated with a reason of something like 'this module needs updating to the current api and may not be functional'

**Comment by Julusian** [comment]
At this point I think its been long enough that it is probably more confusing to do this than not.
And given that it isnt easy to do, I feel that it isnt worth the effort anymore
